src/Auth/CookieAuthenticate.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Auth;
+
+use Cake\Auth\BaseAuthenticate;
+use Cake\Controller\ComponentRegistry;
+use Cake\Controller\Component\CookieComponent;
+use Cake\Controller\Exception\MissingComponentException;
+use Cake\Event\Event;
+use Cake\Network\Request;
+use Cake\Network\Response;
+use Cake\Routing\Router;
+
+/**
+ * Cookie-based adapter for AuthComponent.
+ *
+ * Provides the ability to authenticate using COOKIE
+ *
+ * ```
+ *    $this->Auth->config('authenticate', [
+ *        'Authenticate.Cookie' => [
+ *            'fields' => [
+ *                'identifier' => 'username',
+ *                'token' => 'password'
+ *                'tokenCreated' => null
+ *             ],
+ *            'userModel' => 'Users',
+ *            'scope' => ['Users.active' => 1],
+ *            'crypt' => 'aes',
+ *            'cookie' => [
+ *                'name' => 'RememberMe',
+ *                'time' => '+2 weeks',
+ *            ]
+ *        ]
+ *    ]);
+ * ```
+ */
+class CookieAuthenticate extends BaseAuthenticate
+{
+
+    /**
+     * Constructor
+     *
+     * @param \Cake\Controller\ComponentRegistry $registry The Component registry
+     *   used on this request.
+     * @param array $config Array of config to use.
+     */
+    public function __construct(ComponentRegistry $registry, $config)
+    {
+        $this->_registry = $registry;
+
+        $this->config([
+            'cookie' => [
+                'name' => 'RememberMe',
+                'expires' => '+2 weeks',
+            ],
+            'crypt' => 'aes',
+            'fields' => ['tokenCreated' => null]
+        ]);
+
+        $aliases = [
+            'identifier' => 'username',
+            'token' => 'password'
+        ];
```

Comment:
What if the user records were required to contain a separate column to store a one time use token?